database/migrations/0005-bumps.py

```diff
@@ -0,0 +1,60 @@
+"""
+Create tables for guild bumps and alter player_settings where needed.
+"""
+
+# pylint: disable=invalid-name
+
+from sqlite3 import OperationalError
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from sqlite3 import Connection
+
+
+def run(con: 'Connection'):
+    """
+    Run the migration.
+    """
+    cur = con.cursor()
+
+    cur.execute('''
+        CREATE TABLE IF NOT EXISTS bumps (
+            id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+            idx INTEGER NOT NULL,
+            guild_id INTEGER NOT NULL,
+            title TEXT NOT NULL,
+            author TEXT NOT NULL,
+            url TEXT NOT NULL,
+
+            UNIQUE(guild_id, idx)
+        )
+    ''')
+
+    con.commit()
+
+    try:
+        cur.execute('''
```

Comment:
This is more of a note to self, but in the future we might need to change how we handle column changes in existing schemas. The `try-except` pattern works well so far but I'm wary of it biting us in the future lol

Reply:
@holedaemon Haha, good question. I had some familiarity with SQLAlchemy from past projects that used MariaDB and Postgres, but I've never tried it with SQLite.

Blanco actually started out as a cog from an old general-purpose bot called Rico. While I was spinning Blanco off into a separate bot, I also had to decide on a way to persist state, and a full RDBMS seemed overkill so I decided on SQLite. I read somewhere that SQLite's weak typing and other quirks with `ALTER` statements could affect how ORMs interface with it so I just never bothered with one and wrote migrations by hand.

With 1.0.0 I might finally start using SQLAlchemy. Still think it's a little overkill for a tiny project, but if it helps maintainability then why not.